pkg/kubelet/winstats/version.go

```diff
@@ -33,33 +33,33 @@ type OSInfo struct {
 func GetOSInfo() (*OSInfo, error) {
 	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("open Windows CurrentVersion failed: %w", err)
 	}
 	defer k.Close()
 
 	buildNumber, _, err := k.GetStringValue("CurrentBuildNumber")
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("getOSInfo, get CurrentBuildNumber failed: %w", err)
 	}
 
 	majorVersionNumber, _, err := k.GetIntegerValue("CurrentMajorVersionNumber")
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("getOSInfo, get CurrentMajorVersionNumber failed: %w", err)
 	}
 
 	minorVersionNumber, _, err := k.GetIntegerValue("CurrentMinorVersionNumber")
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("getOSInfo, get CurrentMinorVersionNumber failed: %w", err)
 	}
 
 	revision, _, err := k.GetIntegerValue("UBR")
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("getOSInfo, get UBR failed: %w", err)
 	}
 
 	productName, _, err := k.GetStringValue("ProductName")
 	if err != nil {
-		return nil, nil
+		return nil, fmt.Errorf("getOSInfo, get ProductName failed: %w", err)
```

Comment:
these iterative error messages are definitely adding value. But also I'd suggest to add a whole registry path to the error message so it will be clear from the first error message what is broken. Without the need to iterate over and over giving permissions to one registry branch at a time =)

Reply:
I have added the full path description